src/core/execute.h

```diff
@@ -134,6 +134,7 @@ typedef enum ExecDirectoryType {
 
 typedef struct ExecDirectory {
         char **paths;
+        char **symlinks;
         mode_t mode;
 } ExecDirectory;
```

Comment:
Hmm, this feels a bit redundant/non-normalized to store like this, i.e. all paths are allocated n+1 for n symlinks.

maybe use:

```c
typedef struct ExecDirectoryItem {
        char *name;
        char **symlinks;
} ExecDirectoryItem;
        
typedef struct ExecDirectory {
        mode_t mode;
        size_t n_items;
        ExecDirectoryItem *items;
} ExecDirectory;
```

i.e. store this strictly normalized: for each runtime dir allocate one entry in the `items` directory. this way you'll just allocate each string once, and it can never happen that a symlink without a name is stored.

The make RuntimeDirectory bus property just show the main name, and RuntimeDirectoryWithSymlinks show the same data but with the symlinks list also included.